inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -1076,6 +1076,36 @@ void SubstituteScaleShiftBroadCastPass::run() {
     }
 }
 
+void InsertAdditionalOutputToLSTMCellPass::run() {
```

Comment:
finally got the point.
In lstmcell output of CT is a result of compute layer : LSTMCell3:sum, which in gna2 fuse mode is revealed as scratch buffer, and cannot be extracted.
However implementation does depend on lstm cell specific, while it should rely on outputInfo specific, or even on internal version o this info since we will step in internal modification of this info and not always deliver outstanding outputs to user. 
Since graph might evolve during pass manager work, i think that first or one of first passes should be inserting max_precision_node placeholder.
we can further discuss how to compile that node : for example keep prev layer without acc and use extra diag layer , so we can extract 32b output. Or use simple multiple activaion mode with extra identity inserted.

I think this need to be addressed at least partially in this PR, say having a pass before or as part of insertIdentityActivationFor the layer, that will insert identityActivation for outputNodes as well.
For doing that simple extending isOutputLayer by details from outputInfo, and making corresponding adjustemenets to this pass is simple. We might loose accuracy for having always 16b outputs, but this question to extra Epic i guess.

Reply:
Not sure if that is the way you want to solve it, but I've planned to add:
                auto output_name = output->getName();
                auto is_network_output = outputsMap.find(output_name) != outputsMap.end();
                // In cases that this layer is network output you cannot use identity as sole output on
                // it since it will possibly be fused and layer outputs will be unavailable 
                if(is_network_output) {
                    CNNLayerPtr additional_output =
                        std::make_shared<GenericLayer>(LayerParams({output_name + ":side_identity", "identity", InferenceEngine::Precision::FP32}));

                    additional_output->insData.resize(1);
                    additional_output->outData.resize(1);

                    auto out_data = DataPtr(new Data(output_name + ":side_identity_data", output->getTensorDesc()));
                    getCreatorLayer(out_data) = additional_output;

                    additional_output->outData[0] = out_data;

                    input_to[additional_output->name] = additional_output;
                    additional_output->insData[0] = output;
                }
at the end of InsertIdentityToLSTMCellPass of #2080 PR. Is that the way you've suggested? In #2080 lstm outputs will end with identity, and if that layer is output, then yet another identity will be added, so the first identity is not fused, then both cases when lstmCell is input to another layer + memory, as well as cases where lstmCell is output are covered

Reply:
i think you got my point - lstm cell just one of example on missed outputs - need to generalize solution. But in pass manager we dont have outputs info, so this better to achive by providing outputs info into certain pass like ConfigureOutputLayersPass(outputsInfo) - where you will remove specific you previously did for LSTMCell while concentrate on iterating over outputsInfo

Reply:
General solution created

Reply:
Ok looks fine from glance - can we add more tests, at least 1 more test with addOutput call for every layers or for several layers.

Reply:
Will do for both cases